internal/provider/resource_host_catalog_plugin_test.go

```diff
@@ -0,0 +1,367 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/hostcatalogs"
+	"github.com/hashicorp/boundary/testing/controller"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	_ "github.com/kr/pretty" // So I don't have to keep adding to/removing from go.mod :-)
+)
+
+const (
+	testPluginHostCatalogDescription        = "bar"
+	testPluginHostCatalogDescriptionUpdate  = "bar foo"
+	testPluginHostCatalogDescriptionUpdate2 = "bar foo foo"
+)
+
+type expectedAttributesState uint
+
+const (
+	expectedAttributesStatePreviouslyEmptyNowSet expectedAttributesState = iota
+	expectedAttributesStatePreviouslySetNoChange
```

Comment:
Consider documenting here that for the tests here, when we expect attributes to change, we'll also expect secrets to change.